substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/SecurityServicesFeature.java

```diff
@@ -659,7 +659,15 @@ private static boolean isValid(Service s) {
     private static Function<String, Class<?>> getConstructorParameterClassAccessor(ImageClassLoader loader) {
         Map<String, /* EngineDescription */ Object> knownEngines = ReflectionUtil.readStaticField(Provider.class, "knownEngines");
         Class<?> clazz = loader.findClassOrFail("java.security.Provider$EngineDescription");
-        Field consParamClassNameField = ReflectionUtil.lookupField(clazz, "constructorParameterClassName");
+        Field consParamClassField;
+
+        try {
+            consParamClassField = ReflectionUtil.lookupField(clazz, "constructorParameterClassName");
+        } catch (ReflectionUtil.ReflectionUtilError e) {
+            consParamClassField = ReflectionUtil.lookupField(clazz, "constructorParameterClass");
+        }
+
+        Field consParamClassFieldFinal = consParamClassField;
```

Comment:
`final Field conParamClassFieldFinal = ...`?

Reply:
It's required because it's used in the returned lambda and needs to be effectively final, or `final`.

Reply:
Yes. Mark it `final`?

Reply:
Then it complains about potentially not being initialized due to the try-catch initialization. It feels like I am missing something trivial so please add a code suggestion if so.